packages/components/alert/src/use-alert.ts

```diff
@@ -0,0 +1,123 @@
+import {HTMLNextUIProps, mapPropsVariants, PropGetter} from "@nextui-org/system";
+import {AlertSlots, SlotsToClasses} from "@nextui-org/theme";
+import {ReactRef, useDOMRef} from "@nextui-org/react-utils";
+import {AlertVariantProps} from "@nextui-org/theme/src/components/alert";
+import {ReactNode, useCallback, useMemo} from "react";
+import {alert} from "@nextui-org/theme";
+import {useState} from "react";
+import {objectToDeps} from "@nextui-org/shared-utils";
+
+interface Props extends HTMLNextUIProps<"div"> {
+  /**
+   * Ref to the DOM node.
+   */
+  ref?: ReactRef<HTMLDivElement | null>;
+
+  /**
+   * title of the alert message
+   */
+  title: string;
+
+  /**
+   * Main body of the alert message
+   */
+  description: ReactNode;
+
+  /**
+   *  whether the alert can be closed by user
+   */
+  isCloseable?: boolean;
+
+  /**
+   * function which is called when close button is clicked
+   */
+  onClose?: () => void;
+
+  /**
+   * Classname or List of classes to change the classNames of the element.
+   * if `className` is passed, it will be added to the base slot.
+   *
+   * @example
+   * ```ts
+   * <Alert classNames={{
+   *    base:"base-classes", // image classes
+   *    mainWrapper: "mainWrapper-classes"
+   *    description: "description-classes"
+   *    title: "title-classes"
+   * }} />
+   * ```
+   */
+  classNames?: SlotsToClasses<AlertSlots>;
+}
+export type UseAlertProps = Props & AlertVariantProps;
+
+export function useAlert(originalProps: UseAlertProps) {
+  const [props, variantProps] = mapPropsVariants(originalProps, alert.variantKeys);
+
+  const [isVisible, setIsVisible] = useState(true);
+  const handleClose = () => {
+    setIsVisible(() => false);
+  };
+
+  const {title, description, onClose, isCloseable, ref, classNames} = {
+    /**
+     *  isCloseable is true by default if not provided in props
+     */
+    isCloseable: true,
+
+    /**
+     * By default, onClose simply closes the alert
+     */
+    onClose: handleClose,
```

Comment:
what's the reason defining there here while they are already in `props`? or you just wanna define the default value? if so, you can simply do in this way

```tsx
  const {title, description, onClose = handleClose, isCloseable = true, ref, classNames} = props;
```

Reply:
yeah, I was trying to set defautl values. Will use the syntax you suggested.